octobot_trading/personal_data/positions/types/linear_position.py

```diff
@@ -124,7 +124,8 @@ def update_average_entry_price(self, update_size, update_price):
                                           + (Update quantity * Update price)]
         """
         total_contract_value = self.size + update_size
-        self.entry_price = ((self.size * self.entry_price + update_size * update_price) /
-                            (total_contract_value if total_contract_value != constants.ZERO else constants.ONE))
+        if self._is_update_increasing_size(update_size):
```

Comment:
That's a good fix but I think it should be put in `Position.update_from_order():276` instead of here

Reply:
Right ! changing it